### PR TITLE
update html-minifier version to 3.5

### PR DESCRIPTION
**Pull request by edgesite**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:
If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/kangax/html-minifier
- [x] Increase the version number in the header if appropriate.

What changes in this pr:
- Reorder fields to [html-minifier's docs](https://github.com/kangax/html-minifier#options-quick-reference)
- Add type definitions to features which bring by newer html-minifier

### CODE REVIEW

File: types/html-minifier/index.d.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
-// Type definitions for HTMLMinifier v1.1.1
+// Type definitions for HTMLMinifier v3.5.7
```
Comment (minestarks) [inline types/html-minifier/index.d.ts]: Could you back up the old version to a folder per the readme: https://github.com/DefinitelyTyped/DefinitelyTyped#i-want-to-update-a-package-to-a-new-major-version

**Review by minestarks — CHANGES_REQUESTED** [review]

**Comment by rikuayanokozy** [comment]
Sorry for late! I've duplicate v3 folder and update dependencies in other packages.

File: types/html-minifier/v3/html-minifier-tests.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+
```
Comment (paulvanbrenk) [inline types/html-minifier/v3/html-minifier-tests.ts]: @rikuayanokozy looks like you did the opposite of what @minestarks intended. Generally you would move the old version to a 'Vx' folder, and make the new version the default. So in your case, you would move what's now in DT to 'v1', and keep your changes in the root.

**Review by paulvanbrenk — CHANGES_REQUESTED** [review]

File: types/html-minifier/v3/html-minifier-tests.ts (hunk 2 above)
Comment (rikuayanokozy) [reply types/html-minifier/v3/html-minifier-tests.ts]: 🤣OMG, What am I doing??!
A new commit fix that, please check it.

File: types/html-minifier/v1/tslint.json [hunk 3]
```diff
@@ -0,0 +1,79 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
```
Comment (paulvanbrenk) [inline types/html-minifier/v1/tslint.json]: Do we still need all these rules disabled in the v3 version, can you make a sweep and see if you can clean this up a little?

NOTE!... don't really care about v1, just v3!!

**Review by paulvanbrenk — CHANGES_REQUESTED** [review]

**Comment by rikuayanokozy** [comment]
@paulvanbrenk just made a sweep on that, only the `no-reference-import` left there due to UglifyJS is an optional dependency.
Happy new year!

**Pull request merged**